core/src/main/java/org/apache/iceberg/SnapshotManager.java

```diff
@@ -54,6 +55,57 @@ public ManageSnapshots rollbackTo(long snapshotId) {
     return this;
   }
 
+  @Override
+  public ManageSnapshots createBranch(String name, long snapshotId) {
+    updateSnapshotReferencesOperation()
+        .createBranch(name, snapshotId)
+        .commit();
+    return this;
+  }
+
+  @Override
+  public ManageSnapshots createTag(String name, long snapshotId) {
+    updateSnapshotReferencesOperation().createTag(name, snapshotId).commit();
+    return this;
+  }
+
+  @Override
+  public ManageSnapshots removeBranch(String name) {
+    updateSnapshotReferencesOperation().removeBranch(name).commit();
+    return this;
+  }
+
+  @Override
+  public ManageSnapshots removeTag(String name) {
+    updateSnapshotReferencesOperation().removeTag(name).commit();
+    return this;
+  }
+
+  @Override
+  public ManageSnapshots setMinSnapshotsToKeep(String name, int minSnapshotsToKeep) {
+    updateSnapshotReferencesOperation().setMinSnapshotsToKeep(name, minSnapshotsToKeep);
+    return this;
+  }
+
+  @Override
+  public ManageSnapshots setTagRetention(String name, Long maxRefAgeMs) {
+    updateSnapshotReferencesOperation().setTagRetention(name, maxRefAgeMs);
+    return this;
+  }
+
+  @Override
+  public ManageSnapshots renameBranch(String name, String newName) {
+    updateSnapshotReferencesOperation().renameBranch(name, newName);
+    return this;
+  }
+
+  private UpdateSnapshotReferencesOperation updateSnapshotReferencesOperation() {
```

Comment:
This is good, but it also needs to commit this if there are `cherrypick` or `rollback` calls. Table modifications need to be kept in order, but we want to make as many ref changes as we can simultaneously.

Reply:
Yup, updated! I was thinking if there's a good abstraction here, but I think for now it's simpler just to keep track if there's an ongoing ref operation. In the cherrypick/rollback operations, prior to actually performing those operations we would check if there's an ongoing ref operation and commit it.

For ref metadata chains, a caller would call the commit on the snapshot manager, which would then accomplish our goal for wrapping as many ref changes as we can in a single commit. Let me know if I missed anything in this approach.

Reply:
Updated